src/libraries/Native/Unix/System.Native/pal_networking.c

```diff
@@ -162,6 +162,80 @@ c_static_assert(offsetof(IOVector, Count) == offsetof(iovec, iov_len));
 
 #define Min(left,right) (((left) < (right)) ? (left) : (right))
 
+static bool TryConvertAddressFamilyPlatformToPal(sa_family_t platformAddressFamily, int32_t* palAddressFamily)
+{
+    assert(palAddressFamily != NULL);
+
+    switch (platformAddressFamily)
+    {
+        case AF_UNSPEC:
+            *palAddressFamily = AddressFamily_AF_UNSPEC;
+            return true;
+
+        case AF_UNIX:
+            *palAddressFamily = AddressFamily_AF_UNIX;
+            return true;
+
+        case AF_INET:
+            *palAddressFamily = AddressFamily_AF_INET;
+            return true;
+
+        case AF_INET6:
+            *palAddressFamily = AddressFamily_AF_INET6;
+            return true;
+#ifdef AF_PACKET
+        case AF_PACKET:
+            *palAddressFamily = AddressFamily_AF_PACKET;
+            return true;
+#endif
+#ifdef AF_CAN
+        case AF_CAN:
+            *palAddressFamily = AddressFamily_AF_CAN;
+            return true;
+#endif
+        default:
+            *palAddressFamily = platformAddressFamily;
```

Comment:
I know you're just moving code around, but would be nice if we could assert that no value in `platformAddressFamily` could ever be any `AddressFamily_*` value.

Reply:
Why do we do this assignment at all?  Who uses the written value when the function returns false? (Same question for the other direction below)

Reply:
The value is never used if `false` is returned, so we could remove the assignment. I don't think an `assert` is appropriate, as we do not check for `AddressFamily` validity in BCL